xray/plotting.py

```diff
@@ -310,3 +311,25 @@ def plot_contourf(x, y, z, ax, **kwargs):
     """
     primitive = ax.contourf(x, y, z, **kwargs)
     return ax, primitive
+
+
+@_plot2d
+def plot_pcolormesh(x, y, z, ax, **kwargs):
+    """
+    Pseudocolor plot of 2d DataArray
+
+    Wraps matplotlib.pyplot.pcolormesh
+    """
+    def infer_bounds(coord):
```

Comment:
This may be useful in other places in plotting- could add a docstring and make it a module level function `_infer_bounds`. Example is probably more clear than docstring:
```
In [24]: infer_bounds(np.arange(10))
Out[24]: array([-0.5,  0.5,  1.5,  2.5,  3.5,  4.5,  5.5,  6.5,  7.5,  8.5,  9.5])
```